src/setfit/trainer_distillation.py

```diff
@@ -67,174 +67,137 @@ def __init__(
-            self.student_model.model_body.fit(
-                train_objectives=[(train_dataloader, train_loss)],
-                epochs=num_epochs,
-                optimizer_params={"lr": learning_rate},
-                warmup_steps=warmup_steps,
-                show_progress_bar=show_progress_bar,
-                use_amp=self.use_amp,
-            )
-
-        if not self.student_model.has_differentiable_head or not self._freeze:
-            # Train the final classifier
-            self.student_model.fit(
-                x_train,
-                y_train,
-                num_epochs=num_epochs,
-                batch_size=batch_size,
-                learning_rate=learning_rate,
-                body_learning_rate=body_learning_rate,
-                l2_weight=l2_weight,
-                show_progress_bar=show_progress_bar,
-            )
+                train_loss = self.loss_class(
+                    model=self.student_model.model_body,
+                    distance_metric=args.distance_metric,
+                    margin=args.margin,
+                )
+        else:
+            train_examples = []
+
+            # **************** student training *********************
+            # Only this snippet differs from Trainer.train_embeddings
+            x_train_embd_student = self.teacher_model.model_body.encode(x_train)
+            y_train = self.teacher_model.model_head.predict(x_train_embd_student)
+
+            cos_sim_matrix = util.cos_sim(x_train_embd_student, x_train_embd_student)
+
+            train_examples = []
+            for _ in range(args.num_iterations):
+                train_examples = sentence_pairs_generation_cos_sim(np.array(x_train), train_examples, cos_sim_matrix)
+            # **************** student training END *****************
+
+            batch_size = args.embedding_batch_size
+            train_dataloader = DataLoader(train_examples, shuffle=True, batch_size=batch_size)
+            train_loss = self.loss_class(self.student_model.model_body)
+
+        total_train_steps = len(train_dataloader) * args.embedding_num_epochs
+        logger.info("***** Running training *****")
+        logger.info(f"  Num examples = {len(train_examples)}")
+        logger.info(f"  Num epochs = {args.embedding_num_epochs}")
+        logger.info(f"  Total optimization steps = {total_train_steps}")
+        logger.info(f"  Total train batch size = {batch_size}")
+
+        warmup_steps = math.ceil(total_train_steps * args.warmup_proportion)
+        self.student_model.model_body.fit(
+            train_objectives=[(train_dataloader, train_loss)],
+            epochs=args.embedding_num_epochs,
+            optimizer_params={"lr": args.body_embedding_learning_rate},
+            warmup_steps=warmup_steps,
+            show_progress_bar=args.show_progress_bar,
+            use_amp=args.use_amp,
+        )
+
+
+class DistillationSetFitTrainer(DistillationTrainer):
```

Comment:
Should we add a deprecation warning for this too?

Reply:
We should! I missed this. Thanks